teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -221,13 +221,28 @@ identify related objects.
 
 #### 1. Add Minimal Embedded Status
 
-We will introduce a new struct to hold references to child `TaskRuns` and `Runs`:
+We will introduce a new struct to hold references to child `TaskRuns` and `Runs`, and
+their corresponding `WhenExpressions` and `ConditionChecks`:
 
 ```go
 type childReference struct {
   runtime.TypeMeta           `json:",inline"` // contains API version and Kind
   Name              string   `json:"name,omitempty"` // name of the TaskRun/Run
   PipelineTaskName  string   `json:"pipelineTaskName,omitempty"` // name of the PipelineTask used to create the TaskRun/Run
+  ConditionChecks   []*PipelineRunChildReferenceConditionCheckStatus `json:"conditionChecks,omitempty"` // the condition checks for the TaskRun/Run in the pipeline
+  WhenExpressions []WhenExpression `json:"whenExpressions,omitempty"` // the WhenExpressions for the TaskRun/Run in the pipeline
+}
+```
+
+Additionally, we will add a new struct `PipelineRunChildReferenceConditionCheckStatus`
```

Comment:
I'm a bit confused by this, can't we reuse the existing `PipelineRunConditionCheckStatus` struct?

Reply:
Because `PipelineRunConditionCheckStatus` doesn't have the `ConditionCheckName` as a field - it's currently just used in a map of `ConditionCheckName` to `PipelineRunConditionCheckStatus`, so we need to add a field to it, basically, unless we wanted to use a map instead of an array.

Reply:
Aaaah - we could just use `PipelineRunConditionCheckStatus` and add a new `ConditionCheckName` field to it, leaving it as the empty string when it's being used in the old fully-embedded form.

Reply:
It already has [ConditionName](https://github.com/tektoncd/pipeline/blob/4f8c227794abd6b2157856892415d5c207352295/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L481), is that different from `ConditionCheckName`?

Reply:
It is! `ConditionCheckName` is to a `PipelineTask` name as `ConditionName` is to a `Task` name, basically.

Reply:
got it-- I think I'd prefer something like:

```
type PipelineRunChildConditionCheckStatus {
   ConditionCheckName string
   PipelineRunConditionCheckStatus
}
```

with ChildStatus:
```
type ChildStatus struct {
   ...
   ConditionChecks []PipelineRunChildConditionCheckStatus
}
```
and a note that the `ConditionChecks` field will be removed when support for Conditions is removed.